### PR TITLE
pachsql: ParseURL

### PR DESCRIPTION
**Pull request by brendoncarroll**
Adds a ParseURL function to pachsql.  Renamed package `sql` to `pachsql` to avoid conflict with standard library package name.

### CODE REVIEW

File: src/internal/pachsql/url.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+		return nil, err
+	}
+	port, err := strconv.Atoi(u.Port())
+	if err != nil {
+		return nil, err
+	}
+	params := make(map[string]string)
+	for k, v := range u.Query() {
+		if len(v) > 0 {
+			params[k] = v[len(v)-1]
```
Comment (acohen4) [inline src/internal/pachsql/url.go:32]: do we already know what param keys may have multiple values?

File: src/internal/pachsql/url.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	case "mysql":
+		driver = "mysql"
+		dsn = mySQLDSN(u, password)
+	default:
+		return nil, errors.Errorf("database protocol %q not supported", u.Protocol)
+	}
+	return sqlx.Open(driver, dsn)
+}
+
+func postgresDSN(u URL, password string) string {
```
Comment (acohen4) [inline src/internal/pachsql/url.go]: does postgres not support additional params? Thinking about the unused `u.Params` field in this func

File: src/internal/pachsql/url_test.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+				Host:     "10.0.0.1",
+				Port:     9000,
+				Database: "mydb",
+				Params: map[string]string{
+					"sslmode": "disable",
+				},
+			},
+		},
+		{
+			In: "mysql://jbond@10.0.0.1:1007/martini?shaken=true&stirred=false",
```
Comment (acohen4) [inline src/internal/pachsql/url_test.go:27]: lol <3

File: src/internal/pachsql/db.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	case "mysql":
+		driver = "mysql"
+		dsn = mySQLDSN(u, password)
+	default:
+		return nil, errors.Errorf("database protocol %q not supported", u.Protocol)
+	}
+	return sqlx.Open(driver, dsn)
+}
+
+func postgresDSN(u URL, password string) string {
```
Comment (acohen4) [inline src/internal/pachsql/db.go:35]: Should the URL params also be passed to the PG DSN?

File: src/internal/pachsql/url.go (hunk 1 above)
Comment (brendoncarroll) [reply src/internal/pachsql/url.go:32]: I don't think any of them will have multiple values.  A URL is allowed have multiple values per parameter though.

File: src/internal/pachsql/db.go (hunk 4 above)
Comment (brendoncarroll) [reply src/internal/pachsql/db.go:35]: Yeah, good catch, will add that.

File: src/internal/pachsql/db.go [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+		fields["password"] = password
+	}
+	for k, v := range u.Params {
+		fields[k] = v
+	}
+	var dsnParts []string
+	for k, v := range fields {
+		dsnParts = append(dsnParts, k+"="+v)
+	}
+	return strings.Join(dsnParts, " ")
```
Comment (acohen4) [inline src/internal/pachsql/db.go:52]: Should have asked this before, is the postgres DSN unordered?

**Review by acohen4 — APPROVED** [review]
LGTM!

File: src/internal/pachsql/db.go (hunk 5 above)
Comment (brendoncarroll) [reply src/internal/pachsql/db.go:52]: I'm pretty sure it is.  This section you commented on is taken from `dbutil`

**Pull request merged**